app/src/main/java/de/westnordost/streetcomplete/quests/note_discussion/NoteDiscussionForm.java

```diff
@@ -162,11 +229,61 @@ private void onClickOk()
 
 		Bundle answer = new Bundle();
 		answer.putString(TEXT, noteText);
+		answer.putStringArrayList(IMAGE_PATHS, imagePaths);
 		applyImmediateAnswer(answer);
 	}
 
+	private void takePhoto()
+	{
+		Intent takePhotoIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+		if (takePhotoIntent.resolveActivity(getActivity().getPackageManager()) != null) {
+			try {
+				photoFile = createImageFile();
+			} catch (IOException e) {
+			}
+			if (photoFile != null) {
+				if (Build.VERSION.SDK_INT > 21) {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, getUriForFile(getActivity(), "de.westnordost.streetcomplete.fileprovider", photoFile));
+				} else {
+					takePhotoIntent.putExtra(MediaStore.EXTRA_OUTPUT, Uri.fromFile(photoFile));
+				}
+				startActivityForResult(takePhotoIntent, REQUEST_TAKE_PHOTO);
+			}
+		}
+	}
+
+	@Override
+	public void onActivityResult(int requestCode, int resultCode, Intent data) {
+		if (requestCode == REQUEST_TAKE_PHOTO) {
+			if (resultCode == RESULT_OK)
+			{
+				imageBitmaps.add(BitmapFactory.decodeFile(photoFile.getAbsolutePath()));
+				imagePaths.add(photoFile.toString());
```

Comment:
Doesn't this mean that the whole photo is displayed in the gridview and not just a thumbnail? This could lead to an Out of Memory problem especially with older phones.

Reply:
And how can we avoid this?

Reply:
Subsampling (reading every x-th pixel into memory). BitmapFactory can do that, see here
https://developer.android.com/reference/android/graphics/BitmapFactory.Options.html#inSampleSize
Look also here: https://stackoverflow.com/questions/4916159/android-get-thumbnail-of-image-on-sd-card-given-uri-of-original-image

Reply:
(I can't tell you exactly how to do this, I'd have to research there as well and I guess this is 90% of the work.)

Reply:
But in my first answer, I meant to paste this link:
https://developer.android.com/topic/performance/graphics/load-bitmap.html#load-bitmap

Reply:
Reading a bit more into the issue, I guess if this becomes more complex than a few dozen lines of code, perhaps simply using Picasso could also be an option.

Reply:
I am using now this piece of code and it works as expected... ```ThumbnailUtils.extractThumbnail(BitmapFactory.decodeFile(file.getPath()), width, height);```